### ISSUE TITLE
K.clear_session() increase the performance regarding validation set

### ISSUE THREAD

**Issue by ghost** [keras-team/keras#9671]
I have a siamese network of two Resnet fine-tuning on Imagenet. My code is working fine with respect to the training set. But regarding the validation set, I realized that when I add K.clear_session() it increase the performance a lot. I have no explanation about that. Could you please make me clear that why this function increases the performance of validation set. By the way, I am using Dice as metric for evaluating the performance.
Thanks.

**Comment by Pyrestone** [comment]
If you look at the source code, it calls 
    tf.reset_default_graph()

I assume your current Model is deleted when you clear the session, and re-initialized when running evaluate().

You could check if the validation set performance after running K.clear_session() is similar to that of an untrained model

If so, you have your answer, which is your model overfits and performs worse than an untrained model.
